test.sh

```diff
@@ -47,5 +49,5 @@ rm -rf spec/reports/*
 rm -rf cucumber/api/features/reports/*
 rm -rf cucumber/policy/features/reports/*
 
-bundle exec rake jenkins || true
+bundle exec rake jenkins
```

Comment:
I know you hate `|| true`. But, the way this script and the pipeline are currently written, no test results will get collected when the tests fail.

Reply:
Good eye. I don't hate it, just wish this wasn't the way it worked :) I'll update this PR so it's working again.

Reply:
I think I've found a better way to do this, without requiring us to hijack exit codes. Since this is a Jenkins-specific workaround I think it's better to handle this in Jenkins logic, not in our application code.

Since each stage can also have a `post` block, I think it's better to use 

`post { always { junit ... } }` to capture the reports.

This pattern is mentioned here: https://jenkins.io/doc/book/pipeline/jenkinsfile/#handling-failures. See the updated Jenkinsfile in this PR.

What do you think Alan?

Reply:
In general, this seems like a good approach to me. The only thing that seems a little weird is using `always` in the `post` block. But, I guess, if this is doc'ed as the current best practice, it's a good way to go.